src/pubtools/_marketplacesvm/tasks/community_push/command.py

```diff
@@ -423,7 +423,8 @@ def _upload(
                     )
             except Exception as exc:
                 log.exception("Failed to publish %s: %s", push_item.name, str(exc), stack_info=True)
-                pi = evolve(push_item, state=State.NOTPUSHED)
+                # At this point pi exists so we need to make sure we don't lose anything
+                pi = evolve(pi, state=State.NOTPUSHED)
```

Comment:
Very nice catch.

Just thinking out-loud. Shall we replace `push_item -> pi` also when calling `self.update_rhsm_metadata` or 
`self.cloud_instance(marketplace).upload`?

@jajreidy @JAVGan WDYT?

Reply:
Yeah that makes sense. Might as well have the most recent data.

Reply:
ty